lang/syn/src/parser/accounts/constraints.rs

```diff
@@ -780,6 +814,44 @@ impl<'ty> ConstraintGroupBuilder<'ty> {
         Ok(())
     }
 
+    fn add_realloc(&mut self, c: Context<ConstraintRealloc>) -> ParseResult<()> {
+        if !matches!(self.f_ty, Some(Ty::ProgramAccount(_)))
+            && !matches!(self.f_ty, Some(Ty::Account(_)))
+            && !matches!(self.f_ty, Some(Ty::Loader(_)))
+            && !matches!(self.f_ty, Some(Ty::AccountLoader(_)))
+        {
+            return Err(ParseError::new(
+                c.span(),
+                "realloc must be on an Account, ProgramAccount, or Loader",
```

Comment:
Need to add `AccountLoader` here and perhaps remove the deprecated `ProgramAccount` type?
```suggestion
                "realloc must be on an Account or AccountLoader",
```

Reply:
do you want me to add the deprecation macro on the actual struct definition for `ProgramAccount`?

Reply:
We can do that in a separate change.